pkg/controller/servicetopology/endpoints/endpoints_controller.go

```diff
@@ -82,12 +81,12 @@ func (r *ReconcileServicetopologyEndpoints) InjectClient(c client.Client) error
 }
```

Comment:
we can use this `client.Client` to new `endpointsAdapter` as following:
```
	r.Client = c
        r.endpointsAdapter = adapter.NewEndpointsAdapter(c)
	return nil
```

then we can remove `InjectConfig(cfg *rest.Config)` function.

Reply:
by the way, it seems that `client.Client` field is unused in `ReconcileServicetopologyEndpoints`, so we can remove it.

Reply:
The client.Client is used here: https://github.com/openyurtio/openyurt/blob/c730e6445dd9fbeba9e106a1f0bd367a8fb0ea37/pkg/controller/servicetopology/endpoints/endpoints_controller.go#L126